### PR TITLE
Converted unit tests from stdlib to gocheck

### LINKED ISSUE #26175: Convert unit tests to use gocheck
Right now the integration tests are using gocheck for having things in suites but we also have unit tests that use the stdlib's `testing` package.  

We need to change all the unit tests to use gochecks suites and not use the stdlibs testing package.

Some of the unit test files are:

https://github.com/docker/docker/blob/master/daemon/daemon_linux_test.go
https://github.com/docker/docker/blob/master/daemon/daemon_test.go
https://github.com/docker/docker/blob/master/daemon/daemon_unix_test.go

etc...

### PR DESCRIPTION
**Pull request by michelvocks**
**\- What I did**
Converted stdlib unit tests to gocheck for the whole project except vendor.

**\- How I did it**
Reused already existing examples by the integration tests which are already using the gocheck suite.

**\- How to verify it**

**\- Description for the changelog**

Converted unit tests from stdlib to gocheck.

fixes #26175 

**\- A picture of a cute animal (not mandatory but encouraged)**

Signed-off-by: Michel Vocks michelvocks@gmail.com

### CODE REVIEW

**Comment by michelvocks** [comment]
Currently there is one problem with the tests of volume/store/store_test.go which is 100% related to the gocheck suite. 
I try to find out what is the real cause.

**Comment by michelvocks** [comment]
Fixed volume/store/store_test.go issue. Some tests did not unregister the used volumedrivers.

**Comment by crosbymichael** [comment]
@michelvocks thanks for doing this. I'll review it today!

File: api/client/bundlefile/bundlefile_test.go [hunk 1]
```diff
@@ -8,9 +8,17 @@ import (
 	"testing"
 
 	"github.com/docker/docker/pkg/testutil/assert"
```
Comment (tonistiigi) [inline api/client/bundlefile/bundlefile_test.go]: We should remove `testutil/assert` pkg

File: api/client/container/ps_test.go [hunk 2]
```diff
@@ -40,34 +40,34 @@ func TestBuildContainerListOptions(t *testing.T) {
 		},
 	}
 
-	for _, c := range contexts {
-		options, err := buildContainerListOptions(c.psOpts)
+	for _, ctxs := range contexts {
+		options, err := buildContainerListOptions(ctxs.psOpts)
 		if err != nil {
-			t.Fatal(err)
+			c.Fatal(err)
```
Comment (tonistiigi) [inline api/client/container/ps_test.go]: `c.Assert(err, checker.IsNil)`

File: api/client/container/ps_test.go [hunk 3]
```diff
@@ -40,34 +40,34 @@ func TestBuildContainerListOptions(t *testing.T) {
...
 		}
 
-		if c.expectedAll != options.All {
-			t.Fatalf("Expected All to be %t but got %t", c.expectedAll, options.All)
+		if ctxs.expectedAll != options.All {
+			c.Fatalf("Expected All to be %t but got %t", ctxs.expectedAll, options.All)
```
Comment (tonistiigi) [inline api/client/container/ps_test.go]: `c.Assert(options.All, checker.Equals, ctxs.expectedAll)

**Comment by tonistiigi** [comment]
@michelvocks I'd recommend doing this couple of packages at a time.

**Comment by vdemeester** [comment]
I definitely agree with @tonistiigi 👼

**Comment by michelvocks** [comment]
@tonistiigi Thanks for the feedback. I'll work on this.
@vdemeester I completely agree that it's hard to review this one large commit. Do you have an idea how I should split up this commit? Maybe one commit for every sub-package (daemon, api, volume, system etc.)?

**Comment by michelvocks** [comment]
@tonistiigi I've removed the `testutil/assert` on the first test `api/client/bundlefile/bundlefile_test.go`. Took me quite a while to figure out how to do the asserts. Especially the last multi-line contains match.

**Comment by crosbymichael** [comment]
@michelvocks you could do 1 pull request for each package, i think that will work well

**Comment by vdemeester** [comment]
I'm gonna close this one, see #26393

**Pull request closed without merging**